onnxruntime/core/session/allocator_adapters.cc

```diff
@@ -44,6 +60,34 @@ const OrtMemoryInfo* OrtAllocatorImplWrappingIAllocator::Info() const {
   return &i_allocator_->Info();
 }
 
+std::string OrtAllocatorImplWrappingIAllocator::Stats() const {
+  AllocatorStats stats{};
+
+  ORT_TRY {
+    i_allocator_->GetStats(&stats);
+  }
+  ORT_CATCH(const NotImplementedException& /*e*/) {
+    return {};
+  }
+
```

Comment:
The purpose of the exception was to report status to the user in the first place. Let it propagate up the stack, do not swallow.